server/lib/controller/run_rule_controller.dart

```diff
@@ -57,10 +57,11 @@ class RunRuleController extends ResourceController {
     final userClient = await _googleApi.getUserAccountClient(refreshToken);
 
     // Creates a DV360 client using the user's authenticated client.
-    final dv360 = DisplayVideo360Client(userClient, _dv360BaseUrl);
+    final dv360 =
+        DisplayVideo360Client(userClient, _firestoreClient, _dv360BaseUrl);
 
     // Runs the rule using the DV360 client.
-    await dv360.run(rule);
+    await dv360.run(rule, userId, ruleId);
```

Comment:
Is `ruleId` also existing in `rule`?

Reply:
The `Rule` model currently doesn't store the rule ID, only the `Rule` proto does. Should I add it to the Rule model?

Reply:
I don't have a strong opinion.

If you decide the add `ruleId` to Rule model, you can remove the `ruleId` in `DV360Client.run()`. 

If not, I see that `Rule` model only have one field: `Action`. Do you plan to add other fields in the future? If not, I think you can just use DV360Client.run(ruleProto...) and convert the Action proto to Action object inside `DV360Client.run()`, instead of converting the Rule Proto to an Rule object and only get the Action object from the Rule object.

Reply:
The `Rule` model will also have a `condition` field in a later PR. So I guess, just keep the same for now?